### DISCUSSION TITLE
Lighthouse deck / positioning issues

### DISCUSSION THREAD

**Discussion by orsmolarski** [bitcraze/discussions#715] [Q&A]
Hello, we are having issues with the lighthouse positioning system, I suspect the lighthouse deck. I have not been able to pin point the issue,  looking for some assistance or guidance.

We are attempting to use the Crazyflie with the lighthouse deck and steam VR 2.0 basestations. Our first few usages and setups were very smooth, positioning was precise and the crazyflie flew stably, both in a setup consisting of 2 basestations and a setup of 4 basestations. Since then,the light deck / positioning system is not working / behaving Inconsistently and unexpectedly.

### We observe the following:
- The lighthouse seems to boot as expected: LED sequence is as shown in - https://www.youtube.com/watch?v=QwaJC-cLHSc  

**Console**:

> SYS: ----------------------------
> SYS: Crazyflie 2.1 is up and running!
> SYS: Build 0:0f2f0f07d2c2 (2023.02) CLEAN
> SYS: I am 0x2032353241335019002E0033 and I have 1024KB of flash!
> CFGBLK: v1, verification [OK]
> DECK_CORE: 1 deck(s) found
> DECK_CORE: Calling INIT on driver bcLighthouse4 for deck 0
> IMU: BMI088: Using I2C interface.
> IMU: BMI088 Gyro connection [OK].
> IMU: BMI088 Accel connection [OK]
> IMU: BMP388 I2C connection [OK]
> ESTIMATOR: Using Kalman (2) estimator
> CONTROLLER: Using PID (1) controller
> MTR-DRV: Using brushed motor driver
> SYS: About to run tests in system.c.
> SYS: NRF51 version: 2023.02
> EEPROM: I2C connection [OK].
> STORAGE: Storage check [OK].
> IMU: BMI088 gyro self-test [OK]
> SYS: Self test passed!
> STAB: Wait for sensor calibration...
> SYS: Free heap: 19184 bytes
> LHFL: Lighthouse bootloader version: 2
> LHFL: Bitstream CRC32: 112BC794 [PASS]
> LHFL: Firmware version 6 verified, booting deck!
> STAB: Ready to fly.
> ESTKALMAN: State out of bounds, resetting
> ESTKALMAN: State out of bounds, resetting
> ESTKALMAN: State out of bounds, resetting
> ESTKALMAN: State out of bounds, resetting

- Inconsistent lighthouse system status: 

1.  In most cases the CF w/lighthouse deck is in the FOV of more than one basestation, though the system status is zero (Only Blue LED is on).
_Logs_: _Transition between system states/LED color achieved by rebooting a few times seems random (mostly Blue LED)_
[lh_logs_static_cf_in_fov_2_lh_blue_LED_on.zip](https://github.com/bitcraze/discussions/files/11430292/lh_logs_static_cf_in_fov_2_lh_blue_LED_on.zip)
[lh_logs_cf_static_in_fov_2_lh_orange_LED_on.zip](https://github.com/bitcraze/discussions/files/11430289/lh_logs_cf_static_in_fov_2_lh_orange_LED_on.zip)
[lh_logs_static_cf_in_fov_2_lh_green_LED_on.zip](https://github.com/bitcraze/discussions/files/11430290/lh_logs_static_cf_in_fov_2_lh_green_LED_on.zip)

2. Some cases the system status is non zero (Green LED is on - base station data is sent to estimator), but seems to be stuck/hang - Orange/Green LED stay on when occluding the lighthouse deck from basestations. (Observed when covering the lighthouse deck with hand, turning the drone upside down, and fortified observation by moving the drone  in to another room with no base stations)

3.   Rarely (happened only twice) the when the base station data is being sent to estimation (Green LED on) and attempting to fly the CF drone, we observe a drift in position, and changes in system status (Green LED on to Orange on and the opposite) but the drone keeps drifting and position is not corrected.
_Logs_:
[lighthouse_hover_drift.zip](https://github.com/bitcraze/discussions/files/11430285/lighthouse_hover_drift.zip)

*Note: Logs contain default lighthouse status log block and in addition another log block with more lighthouse data I though might be useful.

**Troubleshooting we tried:**
* Re-flashing the CF FW - multiple times, different versions.
* Different base station setups
* Replaced Deck connectors
* Flying the CF with RC (Xbox controller) - Works as expected.
* Flying with Flow Deck - Works as expected.
* Replacing Battery
* Replacing motors and props.

Any help or insights would be much appreciated, 
Thanks!

**Comment by tobbeanton** [comment]
Without looking into in detail I'm suspecting there could be a hardware issue with the lighthouse deck. I guess you only have one to test with?

**Reply by orsmolarski** [reply]
Unfortunately no, we only have one lighthouse deck.
